# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 459739221a88
# Full Hash: 459739221a884b2b1c2adab42fd19fafe279bd20
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing build bustages
#   
#   This reverts commit ab681e36aeafc21ba62bb4680bdaa6206f02546d.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r fc9986230203 -r 459739221a88 dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Thu Jun 05 00:15:19 2025 +0300
+++ b/dom/ipc/PWindowGlobal.ipdl	Thu Jun 05 00:19:41 2025 +0300
@@ -15,10 +15,10 @@
 include protocol PInProcess;
 include protocol PBrowserBridge;
 include protocol PWebAuthnTransaction;
-include protocol PWebIdentity;
 
 include DOMTypes;
 include ClientIPCTypes;
+include IPCIdentityCredential;
 include NeckoChannelParams;
 include SessionStoreTypes;
 
@@ -37,7 +37,9 @@
 [RefCounted] using mozilla::dom::SessionStoreRestoreData from "mozilla/dom/SessionStoreRestoreData.h";
 using mozilla::dom::CredentialMediationRequirement from "mozilla/dom/CredentialManagementBinding.h";
 using mozilla::dom::IdentityCredentialDisconnectOptions from "mozilla/dom/IdentityCredentialBinding.h";
+using mozilla::dom::IdentityCredentialInit from "mozilla/dom/IdentityCredentialBinding.h";
 using mozilla::dom::IdentityCredentialRequestOptions from "mozilla/dom/IdentityCredentialBinding.h";
+using mozilla::dom::IdentityLoginTargetType from "mozilla/dom/IdentityCredentialBinding.h";
 using mozilla::dom::LoginStatus from "mozilla/dom/LoginStatusBinding.h";
 
 namespace mozilla {
@@ -67,7 +69,6 @@
 {
   manager PBrowser or PInProcess;
   manages PWebAuthnTransaction;
-  manages PWebIdentity;
 
 child:
   async __delete__();
@@ -97,6 +98,8 @@
 
   async RestoreTabContent(nullable SessionStoreRestoreData aData) returns (bool success);
 
+  async NavigateForIdentityCredentialDiscovery(nsCString aUri, IdentityLoginTargetType aType);
+
 both:
   async RawMessage(JSActorMessageMeta aMetadata, ClonedMessageData? aData,
                    ClonedMessageData? aStack);
@@ -204,6 +207,23 @@
 
   async ReloadWithHttpsOnlyException();
 
+  // Used by the Credential Manager API and FedCM to keep the discovery of
+  // a credential abstracted from the content process. This is required because
+  // credentialed requests that are specifically not partitioned are made and
+  // the results must not enter the child process until the user consents via
+  // purpose built UI.
+  async GetIdentityCredential(IdentityCredentialRequestOptions aOptions,
+                              CredentialMediationRequirement aMediationRequirement,
+                              bool aHasUserActivation)
+    returns (IPCIdentityCredential? identityCredential, nsresult rv);
+  async StoreIdentityCredential(IPCIdentityCredential aCredential)
+      returns (nsresult rv);
+  async DisconnectIdentityCredential(IdentityCredentialDisconnectOptions aOptions) returns (nsresult rv);
+  async PreventSilentAccess() returns (nsresult rv);
+
+  // Indicate that the page has reported itself as logged in
+  async SetLoginStatus(LoginStatus foo) returns (nsresult rv);
+
   async GetStorageAccessPermission(bool aIncludeIdentityCredential) returns(uint32_t permission_action);
 
 
@@ -222,7 +242,6 @@
   async RecordUserActivationForBTP();
 
   async PWebAuthnTransaction();
-  async PWebIdentity();
 
 child:
   async NotifyPermissionChange(nsCString type, uint32_t permission);